lib/console/hardware_identifiers/hardware_identifiers.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Console.HardwareIdentifiers do
+  import Ecto.Query, warn: false
+  alias Console.Repo
+
+  alias Console.HardwareIdentifiers.HardwareIdentifier
+
+  def get_resource_by_hardware_identifier(token, resource) do
+    case Base.decode32(token) do
+      {:ok, token} ->
+        case Repo.get_by(HardwareIdentifier, token: token) do
+          hardwareIdentifier = %HardwareIdentifier{} -> hardwareIdentifier |> Repo.preload(resource)
```

Comment:
i think you could probably do this with a single query (there are 2 trips to the db/Repo in this function)